### PR TITLE
[FIXED JENKINS-44884] Add @Symbol annotations

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-44884](https://issues.jenkins-ci.org/browse/JENKINS-44884)

@reviewbybees

### CODE REVIEW

**Comment by stephenc** [comment]
Downstream of https://github.com/jenkinsci/structs-plugin/pull/21

File: src/main/java/jenkins/scm/impl/trait/WildcardSCMHeadFilterTrait.java [hunk 1]
```diff
@@ -129,6 +130,7 @@ private String getPattern(String branches) {
     /**
      * Our descriptor.
      */
+    @Symbol("headsWildcardFilter")
```
Comment (jglick) [inline src/main/java/jenkins/scm/impl/trait/WildcardSCMHeadFilterTrait.java]: `headWildcardFilter` by analogy with `headRegexFilter` and following the general English rule (as I understand it) that this usage of “heads” is a kind of unmarked genitive

File: src/test/java/jenkins/scm/impl/SymbolAnnotationsTest.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    @Test
+    public void given__mockScm__when__uninstantiating__then__noRaw$class() throws Exception {
+        MockSCMController c = MockSCMController.create();
+        try {
+            c.createRepository("test");
+            MockSCM instance = new MockSCM(c, "test", new MockSCMHead("master"), new MockSCMRevision(
+                    new MockSCMHead("master"), c.getRevision("test", "master")));
+            assertThat(DescribableModel.uninstantiate2_(instance).toString(), allOf(
+                    startsWith("@"),
```
Comment (jglick) [inline src/test/java/jenkins/scm/impl/SymbolAnnotationsTest.java]: This use `TreeMap`s internally, so you can just assert an exact form for the `toString` and it should be stable.

Comment (jglick) [inline src/test/java/jenkins/scm/impl/SymbolAnnotationsTest.java]: Now that you have added the `is` match, the `allOf` matches seem redundant. (Same for other tests.)

**Comment by stephenc** [comment]
> Now that you have added the `is` match, the `allOf` matches seem redundant. (Same for other tests.)

Actually I see the `is` match as redundant and the `allOf` is the actual criteria. If the `is` match breaks then we can understand what the parameters are in fixing the expected string value

**Pull request merged**